File: readkeysqr/src/main/java/com/example/readkeysqr/ReadKeySqr.java
package com.example.readkeysqr;

import java.nio.ByteBuffer;

public class ReadKeySqr {
    static {
        System.loadLibrary("jni-read-keysqr");
    }

    public static native String HelloFromOpenCV();
    /**
     * 
     * 
     * byteBufferForGrayscaleChannel must be a DirectByteBuffer (not just any old ByteBuffer)
     */
    public static native String ReadKeySqrJson(int width, int height, int bytesPerRow, ByteBuffer byteBufferForGrayscaleChannel);
}

File: readkeysqr/src/main/java/com/example/readkeysqr/ReadKeySqr.kt
package com.example.readkeysqr

import java.nio.ByteBuffer

public object ReadKeySqr {
    init {
        System.loadLibrary("jni-read-keysqr")
    }

    external fun HelloFromOpenCV(): String
    external fun ReadKeySqrJson(
            width: Int,
            height: Int,
            bytesPerRow: Int,
            byteBufferForGrayscaleChannel: ByteBuffer
    ): String
}
